src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { ChangeEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { User } from './components/types/User';
+import { Todo } from './components/types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+const getLastTodoId = (prevTodos: Todo[]) => {
+  const findLastId = Math.max(...prevTodos.map(todo => todo.id));
+
+  return findLastId + 1;
+};
+
+export const App: React.FC = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isGoodToSubmit, setIsGoodToSubmit] = useState(false);
+  const [selectedTodos, setSelectedTodos] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!userId || !title.trim()) {
+      setIsGoodToSubmit(true);
+    }
+
+    if (userId && title.trim()) {
```

Comment:
Instead of checking if `userId` exists, it is better to check if `user` exists after calling `getUserById(userId)` because it ensures that a valid user is selected. Change the condition to use `user` object. 🕵️